org.adempiere.base/src/org/compiere/util/DB.java

```diff
@@ -1457,6 +1457,64 @@ public static String getSQLValueStringEx (String trxName, String sql, Object...
     	}
     	return retValue;
     }
+    
+    /**
+     * get concatenated string values from sql
+     * @param trxName optional transaction name
+     * @param sql
+     * @param params list of parameters
+     * @return first value or null
+     * @throws DBException if there is any SQLException
+     */
+    public static String getSQLValueStringConcatMultipleResultsEx (String trxName, String sql, List<Object> params)
+    {
+    	StringBuilder retValue = new StringBuilder();
+    	PreparedStatement pstmt = null;
+    	ResultSet rs = null;
+    	Connection conn = null;
+    	if (trxName == null)
+    		conn = DB.createConnection(true, Connection.TRANSACTION_READ_COMMITTED);

```

Comment:
This is not recommended, if you need to use not auto commit connection, just start a new transaction. Anyway, it seems unnecessary here since this is just a read operation.

Reply:
This seems duplicated from `getSQLArrayObjectsEx` and I've wrote that. Sorry for the poor implementation I've wrote that time, it should use local Trx and use `trx.getConnnection().setReadonly()` instead.